pkg/clickhouse/source.go

```diff
@@ -102,14 +102,56 @@ func toStringSlice(iSlice []interface{}) []string {
 	return values
 }
 
-func (s Source) WriteChunk(_ string, _ io.Reader) error {
-	// TODO
-	return errors.New("not implemented")
+func (s Source) WriteChunk(_ string, r io.Reader) error {
+	ct, err := s.ColumnTypes()
+	if err != nil {
+		return err
+	}
+	tx, err := s.db.Begin()
```

Comment:
Is `tx` thread safe? Better to start a single transaction for the whole import instead of for each chunk.